numba/core/funcdesc.py

```diff
@@ -96,9 +96,7 @@ def lookup_module(self):
                 # ensure module exist
                 return importlib.import_module(self.modname)
             except ImportError:
-                raise ModuleNotFoundError(
-                    f"can't compile {self.qualname}: "
-                    f"import of module {self.modname} failed") from None
+                return _dynamic_module
```

Comment:
With this change, the doc string needs to be updated to remove `Raise exception if the module can't be found.` and update it with the new behavior.

Reply:
Done :)